README.md

```diff
@@ -10,36 +9,139 @@ license](APACHE-2.0.txt) and documentation is provided under the [CC-BY-4.0
+and the [Apache Lucene][JG_LUCENE] indexing backend
 
-The `push-images.sh` script will push the docker images for all the versioned folders in the repo.
+```bash
+$ docker run --name janusgraph-default janusgraph/janusgraph:latest
+```
 
-Prior to publishing, you'll need to login to [Docker Hub][DH] using the `docker login` command.
+**Connect to a JanusGraph server instance from Gremlin Console**
+
+Start another JanusGraph container and connect to the server instance remotely using Gremlin Console
+
+```bash
+$ docker run --rm --link janusgraph-default:janusgraph -it janusgraph/janusgraph:latest ./bin/gremlin.sh
+
+         \,,,/
+         (o o)
+-----oOOo-(3)-oOOo-----
+plugin activated: janusgraph.imports
+plugin activated: tinkerpop.server
+plugin activated: tinkerpop.utilities
+plugin activated: tinkerpop.hadoop
+plugin activated: tinkerpop.spark
+plugin activated: tinkerpop.tinkergraph
+gremlin> :remote connect tinkerpop.server conf/remote.yaml
+==>Configured janusgraph/172.17.0.2:8182
+gremlin> :> g.addV('person').property('name', 'chris')
+==>v[4160]
+gremlin> :> g.V().values('name')
+==>chris
+```
+
+### Using Docker Compose ###
+
+Start a JanusGraph server instance using [docker-compose.yml](docker-compose.yml)
+
+```bash
+$ docker-compose up
+```
+
+Start another JanusGraph container and run Gremlin Console using [docker-compose.yml](docker-compose.yml)
+
+```bash
+$ docker-compose run --rm janusgraph ./bin/gremlin.sh
+```
 
 ## Configuration ##
 
-The docker containers are configured through environment variables. The environment variables are 
-structured to reflect the different configuration properties available in the 
-[JanusGraph Configuration][JG_CONFIG] .
+JanusGraph Docker containers are configured through environment variables, which can be used to
+specify the base configuration template as well as individual configuration items. The common
+usage will be to specify a template for the general environment (e.g. cql-es) and then provide
+additional individual configuration to override/extend the template.
+
+The `JANUS_PROPS_TEMPLATE` environment variable is used to specify the JanusGraph configuration
+template. Properties in the template are used unless overridden individually as described below.
+The available templates depend on the JanusGraph version.
+
+| JANUS_PROPS_TEMPLATE | Supported Versions |
+| -------------------- | ------------------ |
+| berkeleyje (default) | all                |
+| berkeleyje-es        | all                |
+| cassandra-es         | all                |
+| cql-es               | >=0.2.1            |
+
+Individual JanusGraph configuration items can be configured using corresponding environment
+variables. Provided configuration items will be used to override/extend the template. See
+[JanusGraph Configuration][JG_CONFIG] for more information on available configuration options.
+Gremlin server configuration items can also be specified in the environment by using the
```

Comment:
A link to [config options for the Gremlin Server](http://tinkerpop.apache.org/docs/current/reference/#_configuring_2) could also be helpful here.